src/main/scala/workflow/graph/PipelineEnv.scala

```diff
@@ -30,6 +30,14 @@ class PipelineEnv {
   def setOptimizer(optimizer: Optimizer): Unit = {
     _optimizer = optimizer
   }
+
+  /**
+   * Reset this PipelineEnv (clear state and set the Optimizer to the DefaultOptimizer)
+   */
+  def reset(): Unit = {
+    state.clear()
+    setOptimizer(DefaultOptimizer)
```

Comment:
It resets the Optimizer to the default. This is because some unit tests change the optimizer to test different rules or validate things about the execution, and we eventually have to change it back for other unit tests.